File: src/main/java/dao/Select.java
package dao;

import mapper.UserMapper;
import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;
import org.springframework.stereotype.Component;
import pojo.*;

import java.util.List;

/**
 * @author Kayll
 * @date 2018/11/2 20:32
 */
@Component
public class Select implements DatabaseOperate {
    @Override
    public boolean loginForRegist(String account, String password) {
        return false;
    }

    @Override
    public boolean loginForMain(String account, String password) {
        return false;
    }

    @Override
    public StaffWork selectStaffWork(String email){
        StaffWork staffWork ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        staffWork = userMapper.selectStaffWork(email);
        return staffWork;
    }

    @Override
    public int updateStaffPersonal(StaffPersonal staffPersonal) {
        return 0;
    }

    @Override
    public int insertDeskChange(DeskChange deskChange) {
        return 0;
    }


    public List<DeskChange> selectDeskChange(String workid){
        List<DeskChange> deskChange ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        deskChange = userMapper.selectDeskChange(workid);
        return deskChange;
    }
    public List<DeskChange> selectDeskChange2(String workid){
        List<DeskChange> deskChange ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        deskChange = userMapper.selectDeskChange2(workid);
        return deskChange;
    }

    public List<Resign> selectResign(String workid){
        List<Resign> resigns ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        resigns = userMapper.selectResign(workid);
        return resigns;
    }
    public List<Resign> selectResign2(String workid){
        List<Resign> resigns ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        resigns = userMapper.selectResign2(workid);
        return resigns;
    }

    public List<Appeal> selectAppeal(String workid){
        List<Appeal> appeals ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        appeals = userMapper.selectAppeal(workid);
        return appeals;
    }
    public List<Appeal> selectAppeal2(String workid){
        List<Appeal> appeals ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        appeals = userMapper.selectAppeal2(workid);
        return appeals;
    }

    public List<Notice> selectNotice(){
        List<Notice> notice ;
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        notice = userMapper.selectNotice();
        return notice;
    }

    public Notice selectNoticeContent(int id){
        ApplicationContext applicationContext = new ClassPathXmlApplicationContext("spring/applicationContext.xml");
        UserMapper userMapper = applicationContext.getBean(UserMapper.class);
        return userMapper.selectNoticeContent(id);
    }
}

File: src/main/java/dao/Select.kt
package dao

import mapper.UserMapper
import org.springframework.context.ApplicationContext
import org.springframework.context.support.ClassPathXmlApplicationContext
import org.springframework.stereotype.Component
import pojo.*

/**
 * @author Kayll
 * @date 2018/11/2 20:32
 */
@Component
class Select : DatabaseOperate {
    override fun loginForRegist(account: String, password: String): Boolean {
        return false
    }

    override fun loginForMain(account: String, password: String): Boolean {
        return false
    }

    override fun selectStaffWork(email: String): StaffWork {
        val staffWork: StaffWork
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        staffWork = userMapper.selectStaffWork(email)
        return staffWork
    }

    override fun updateStaffPersonal(staffPersonal: StaffPersonal): Int {
        return 0
    }

    override fun insertDeskChange(deskChange: DeskChange): Int {
        return 0
    }


    fun selectDeskChange(workid: String): List<DeskChange> {
        val deskChange: List<DeskChange>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        deskChange = userMapper.selectDeskChange(workid)
        return deskChange
    }

    fun selectDeskChange2(workid: String): List<DeskChange> {
        val deskChange: List<DeskChange>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        deskChange = userMapper.selectDeskChange2(workid)
        return deskChange
    }

    fun selectResign(workid: String): List<Resign> {
        val resigns: List<Resign>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        resigns = userMapper.selectResign(workid)
        return resigns
    }

    fun selectResign2(workid: String): List<Resign> {
        val resigns: List<Resign>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        resigns = userMapper.selectResign2(workid)
        return resigns
    }

    fun selectAppeal(workid: String): List<Appeal> {
        val appeals: List<Appeal>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        appeals = userMapper.selectAppeal(workid)
        return appeals
    }

    fun selectAppeal2(workid: String): List<Appeal> {
        val appeals: List<Appeal>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        appeals = userMapper.selectAppeal2(workid)
        return appeals
    }

    fun selectNotice(): List<Notice> {
        val notice: List<Notice>
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        notice = userMapper.selectNotice()
        return notice
    }

    fun selectNoticeContent(id: Int): Notice {
        val applicationContext = ClassPathXmlApplicationContext("spring/applicationContext.xml")
        val userMapper = applicationContext.getBean(UserMapper::class.java)
        return userMapper.selectNoticeContent(id)
    }
}
